service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/GpcRequestBuilder.java

```diff
@@ -0,0 +1,139 @@
+package uk.nhs.adaptors.gp2gp.gpc;
+
+import static java.lang.String.valueOf;
+
+import static org.apache.http.protocol.HTTP.CONTENT_LEN;
+import static org.apache.http.protocol.HTTP.CONTENT_TYPE;
+import static org.apache.http.protocol.HTTP.TARGET_HOST;
+
+import java.util.Collections;
+
+import javax.net.ssl.SSLException;
+
+import org.hl7.fhir.dstu3.model.BooleanType;
+import org.hl7.fhir.dstu3.model.Identifier;
+import org.hl7.fhir.dstu3.model.IntegerType;
+import org.hl7.fhir.dstu3.model.Parameters;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ReactiveHttpOutputMessage;
+import org.springframework.http.client.reactive.ReactorClientHttpConnector;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.BodyInserter;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.ExchangeStrategies;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import ca.uhn.fhir.parser.IParser;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import reactor.netty.http.client.HttpClient;
+import uk.nhs.adaptors.gp2gp.ehr.EhrExtractStatus;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@Slf4j
+public class GpcRequestBuilder {
+
+    private static final String NHS_NUMBER_SYSTEM = "https://fhir.nhs.uk/Id/nhs-number";
+    private static final String FHIR_CONTENT_TYPE = "application/fhir+json";
+    private static final String SSP_FROM = "Ssp-From";
+    private static final String SSP_TO = "Ssp-To";
+    private static final String SSP_INTERACTION_ID = "Ssp-InteractionID";
+    private static final String SSP_TRACE_ID = "Ssp-TraceID";
+    private static final String AUTHORIZATION = "Authorization";
+    private static final String AUTHORIZATION_BEARER = "Bearer ";
+    private static final int BYTE_COUNT = 16 * 1024 * 1024;
+    private static final int NUMBER_OF_RECENT_CONSULTANTS = 3;
+
+    private final IParser fhirParser;
+    private final GpcTokenBuilder gpcTokenBuilder;
+    private final GpcConfiguration gpcConfiguration;
+    private final GpcWebClientFilter gpcWebClientFilter;
+
+    public Parameters buildGetStructuredRecordRequestBody(GetGpcStructuredTaskDefinition structuredTaskDefinition) {
+        return new Parameters()
+            .addParameter(new Parameters.ParametersParameterComponent()
```

Comment:
I am wondering if we could create a method which would take the name and the value as parameters to reduce the size of this initialisation.